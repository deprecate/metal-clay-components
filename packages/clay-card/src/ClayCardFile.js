import './ClayCardBase';
import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './ClayCardFile.soy.js';

/**
 * Metal ClayCardFile component.
 */
class ClayCardFile extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayCardFile.STATE = {
	/**
	 * Actions in the dropdown on card.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?array|undefined}
	 * @default undefined
	 */
	actionItems: Config.arrayOf(
		Config.shapeOf({
			label: Config.string().required(),
			separator: Config.bool().value(false),
			url: Config.string(),
		})
	),

	/**
	 * Flag to indicate if the card is disabled or not.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?bool|undefined}
	 * @default false
	 */
	disabled: Config.bool().value(false),

	/**
	 * CSS classes to be applied to the element.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default undefined
	 */
	elementClasses: Config.string(),

	/**
	 * Render ClayIcon in the background element.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default documents-and-media
	 */
	icon: Config.string().value('documents-and-media'),

	/**
	 * Id to be applied to the element.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default undefined
	 */
	id: Config.string(),

	/**
	 * Name to be applied to the input element.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default undefined
	 */
	inputName: Config.string(),

	/**
	 * Value to be applied to the input element.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default undefined
	 */
	inputValue: Config.string(),

	/**
	 * Render label in the ClayLabel element.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default undefined
	 */
	label: Config.string(),

	/**
	 * Style of the ClayLabel.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string}
	 * @default secondary
	 */
	labelStyle: Config.oneOf([
		'danger',
		'info',
		'secondary',
		'success',
		'warning',
	]).value('secondary'),

	/**
	 * Flag to indicate if the card is selectable or not.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?bool|undefined}
	 * @default false
	 */
	selectable: Config.bool().value(false),

	/**
	 * Flag to indicate if the card is selected or not.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?bool|undefined}
	 * @default false
	 */
	selected: Config.bool().value(false),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {!string}
	 * @default undefined
	 */
	spritemap: Config.string().required(),

	/**
	 * Subtitle of the card.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default undefined
	 */
	subtitle: Config.string(),

	/**
	 * Title of the card.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default undefined
	 */
	title: Config.string(),

	/**
	 * Render label in the ClaySticker element.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string|undefined}
	 * @default undefined
	 */
	type: Config.string(),

	/**
	 * Style of the ClaySticker element.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string}
	 * @default primary
	 */
	typeStyle: Config.oneOf([
		'danger',
		'dark',
		'info',
		'light',
		'primary',
		'secondary',
		'success',
		'warning',
	]).value('primary'),

	/**
	 * Transform title in the url.
	 * @instance
	 * @memberof ClayCardFile
	 * @type {?string}
	 * @default undefined
	 */
	url: Config.string(),
};

defineWebComponent('clay-card-file', ClayCardFile);

Soy.register(ClayCardFile, templates);

export {ClayCardFile};
export default ClayCardFile;
