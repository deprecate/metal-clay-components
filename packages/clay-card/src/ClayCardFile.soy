{namespace ClayCardFile}

{template .render}
    {@param spritemap: string}
    {@param? actionsItems: list<[
        label: string,
        separator: bool,
        url: string
    ]>}
    {@param? checkbox: [
		checked: bool,
		disabled: bool,
		indeterminate: bool,
		name: string,
		value: string
	]}
    {@param? elementClasses: string}
    {@param? icon: string}
    {@param? id: string}
    {@param? label: string}
	{@param? labelStyle: string}
    {@param? subtitle: string}
    {@param? title: string}
    {@param? type: string}
	{@param? typeStyle: string}
    {@param? url: string}

    {call ClayCardBase.render}
        {param actionsItems: $actionsItems /}
        {param checkbox: $checkbox /}
        {param elementClasses: $elementClasses /}
        {param icon: $icon /}
        {param id: $id /}
        {param label: $label /}
        {param labelStyle: $labelStyle /}
        {param ref: 'cardFile' /}
        {param spritemap: $spritemap /}
        {param subtitle: $subtitle /}
        {param title: $title /}
        {param type: $type /}
        {param typeStyle: $typeStyle /}
        {param url: $url /}
        {param variant: 'file' /}
    {/call}
{/template}
