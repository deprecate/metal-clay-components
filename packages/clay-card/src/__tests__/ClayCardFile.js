import ClayCardFile from '../ClayCardFile';

let component;
let spritemap = 'icons.svg';
let actionItems = [
	{
		label: 'Edit',
		url: '#1',
	},
	{
		label: 'Save',
		url: '#1',
	},
];

describe('ClayCardFile', function() {
	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render the default markup', () => {
		component = new ClayCardFile({
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with classes', () => {
		component = new ClayCardFile({
			spritemap: spritemap,
			elementClasses: 'my-custom-class',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with id', () => {
		component = new ClayCardFile({
			spritemap: spritemap,
			id: 'myId',
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with actionItems', () => {
		component = new ClayCardFile({
			actionItems: actionItems,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a selected ClayCardFile', () => {
		component = new ClayCardFile({
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a disabled ClayCardFile', () => {
		component = new ClayCardFile({
			disabled: true,
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with input `name`', () => {
		component = new ClayCardFile({
			inputName: 'checkbox01',
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with input `value`', () => {
		component = new ClayCardFile({
			inputValue: 'checkbox',
			selectable: true,
			selected: true,
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with icon', () => {
		component = new ClayCardFile({
			icon: 'documents',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with label', () => {
		component = new ClayCardFile({
			labels: [
				{
					label: 'Approved',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with labelStyle `danger color`', () => {
		component = new ClayCardFile({
			labels: [
				{
					label: 'Approved',
					style: 'danger',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with labelStyle `info color`', () => {
		component = new ClayCardFile({
			labels: [
				{
					label: 'Approved',
					style: 'info',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with labelStyle `secondary color`', () => {
		component = new ClayCardFile({
			labels: [
				{
					label: 'Approved',
					style: 'secondary',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with labelStyle `success color`', () => {
		component = new ClayCardFile({
			labels: [
				{
					label: 'Approved',
					style: 'success',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with labelStyle `warning color`', () => {
		component = new ClayCardFile({
			labels: [
				{
					label: 'Approved',
					style: 'warning',
				},
			],
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with subtitle', () => {
		component = new ClayCardFile({
			subtitle: 'Author Action',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with title', () => {
		component = new ClayCardFile({
			title: 'empty-background.png',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type', () => {
		component = new ClayCardFile({
			type: 'DOC',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type `danger color`', () => {
		component = new ClayCardFile({
			type: 'DOC',
			typeStyle: 'danger',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type `dark color`', () => {
		component = new ClayCardFile({
			type: 'DOC',
			typeStyle: 'dark',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type `info color`', () => {
		component = new ClayCardFile({
			type: 'DOC',
			typeStyle: 'info',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type `light color`', () => {
		component = new ClayCardFile({
			type: 'DOC',
			typeStyle: 'light',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type `primary color`', () => {
		component = new ClayCardFile({
			type: 'DOC',
			typeStyle: 'primary',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type `secondary color`', () => {
		component = new ClayCardFile({
			type: 'DOC',
			typeStyle: 'secondary',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type `success color`', () => {
		component = new ClayCardFile({
			type: 'DOC',
			typeStyle: 'success',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with type `warning color`', () => {
		component = new ClayCardFile({
			type: 'DOC',
			typeStyle: 'warning',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});

	it('should render a ClayCardFile with url', () => {
		component = new ClayCardFile({
			url: '#1',
			spritemap: spritemap,
		});

		expect(component).toMatchSnapshot();
	});
});
