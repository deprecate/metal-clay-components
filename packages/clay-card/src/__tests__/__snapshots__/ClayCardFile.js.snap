// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayCardFile should render a ClayCardFile with actionItems 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
      <div class="flex-col">
        <div class="dropdown dropdown-action">
          <button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v">
              <use xlink:href="icons.svg#ellipsis-v"></use>
            </svg>
          </button>
          <div class="dropdown-menu">
            <ul class="list-unstyled">
              <li>
                <a class=" dropdown-item" href="#1">Edit</a>
              </li>
              <li>
                <a class=" dropdown-item" href="#1">Save</a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with checkbox \`name\` 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" name="checkbox01" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first"></div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with checkbox \`value\` 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" value="checkbox" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first"></div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with checkbox checked 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first"></div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with checkbox disabled 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" disabled="disabled" class="custom-control-input" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first"></div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with checkbox indeterminate 1`] = `
<div class="form-check form-check-card form-check-top-left card-type-asset">
  <div class="custom-control custom-checkbox">
    <label>
      <input checked="" class="custom-control-input" ref="input" type="checkbox">
      <span class="custom-control-indicator"></span>
      <div class="card card-type-asset">
        <div class="aspect-ratio card-item-first"></div>
        <div class="card-body">
          <div class="card-row">
            <div class="flex-col flex-col-expand"></div>
          </div>
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with classes 1`] = `
<div class="card card-type-asset my-custom-class">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with icon 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <div class="aspect-ratio-item-center-middle aspect-ratio-item-fluid">
      <svg aria-hidden="true" class="lexicon-icon lexicon-icon-documents">
        <use xlink:href="icons.svg#documents"></use>
      </svg>
    </div>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with id 1`] = `
<div class="card card-type-asset" id="myId">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with label 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-secondary">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with labelStyle \`danger color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-danger">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with labelStyle \`info color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-info">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with labelStyle \`secondary color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-secondary">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with labelStyle \`success color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-success">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with labelStyle \`warning color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-detail">
          <span class="label label-warning">Approved</span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with subtitle 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-subtitle text-truncate" title="Author Action">Author Action</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with title 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <div class="card-title text-truncate" title="empty-background.png">empty-background.png</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type \`danger color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-danger sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type \`dark color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-dark sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type \`info color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-info sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type \`light color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-light sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type \`primary color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-primary sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type \`secondary color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-secondary sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type \`success color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-success sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type \`warning color\` 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-warning sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with type 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first">
    <span class="sticker sticker-primary sticker-bottom-left">DOC</span>
  </div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render a ClayCardFile with url 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand">
        <a class=" card-title text-truncate" href="#1"></a>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayCardFile should render the default markup 1`] = `
<div class="card card-type-asset">
  <div class="aspect-ratio card-item-first"></div>
  <div class="card-body">
    <div class="card-row">
      <div class="flex-col flex-col-expand"></div>
    </div>
  </div>
</div>
`;
